Payment token fix

harrymoy opened the pull request:
Using the ERC-20 method for the transfer of Dai to pay for the bond. Note that the approval for ERC-20 must come from the Dai address, handled in the React code.

harrymoy committed:
ERC-20 Usage

Using the ERC-20 method for the transfer of Dai to pay for the bond. Note that the approve for ERC-20 must come from the Dai address, handled in the React code

harrymoy committed:
Emitting address

Emitting the address of the bond rather than ID as it'd provide a better user experience (ie they can look it up on Etherscan).

samclusker reviewed (commented):
Can the emit of the BondWallet bond address be followed straight away by the subscribeToBond call? Will the DAI approve function happen in time?

harrymoy committed:
Separated the approve and added documentation

A new function has been created to request approval for the bond. This function should be called first when subscribing to get approval, it returns the address of the bond that gets passed to the Dai.approve() function to allow the bond to make the transfer. After that, the subscriberToBond() function can be called.

harrymoy committed:
More documentation

harrymoy committed:
Merge branch 'dev' into paymentTokenFix
